primitives/ethereum-poa/src/lib.rs

```diff
@@ -436,6 +458,25 @@ pub fn public_to_address(public: &[u8; 64]) -> Address {
 	result
 }
 
+/// Return author's signature over given message.
+#[cfg(feature = "test-helpers")]
```

Comment:
Instead of adding a bunch of feature-gated methods to `src/lib.rs` you can move everything to a separate module and feature-gate it.

That will require decoupling `sign_by` and `sign_by_set` from `Header`, but it's okay imho. If really needed we can extend `Header` by adding an extension trait.